lib/blocs/DetailBloc/detail_bloc.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:moovup_demo/config/environment.dart';
+import 'package:moovup_demo/helpers/api.dart';
+import 'package:moovup_demo/repositories/job_post.dart';
+import 'package:moovup_demo/services/GraphQLService.dart';
+
+import 'detail_events.dart';
+import 'detail_states.dart';
+
+class DetailBloc extends Bloc<DetailEvents, DetailStates> {
+  late GraphQLService service = GraphQLService();
+  late PostRepository repository;
+
+  DetailBloc(GraphQLService _graphQLService) : super(DetailStates()) {
+    this.service = GraphQLService();
+    this.repository = PostRepository(client: service);
+  }
+
+  DetailStates get initialState => OnLoading();
+
+  @override
+  Stream<DetailStates> mapEventToState(DetailEvents event) async* {
+    if (event is FetchDetailData) {
+      yield* _mapFetchDetailDataToStates(event);
+    }
+  }
+
+  Stream<DetailStates> _mapFetchDetailDataToStates(FetchDetailData event) async* {
+    // final query = event.query;
+    // final variables = event.variables;
```

Comment:
if code is not be used, please remove it. Or u can mark as the reference 
for instance: 
//MARK: below code will be using in XXXX feature